src/pages/development/backward-incompatible-changes/highlights.md

```diff
@@ -42,6 +43,25 @@ The following module is affected by this change:
 
 * [Magento_Sales](/module-reference/module-sales/)
 
+### MySQL 8.4 foreign key restrictions
+
+MySQL 8.4 introduces stricter foreign key validation by default. The `restrict_fk_on_non_standard_key` setting is now ON by default, which restricts the use of non-unique or partial keys as foreign keys. This change affects merchants upgrading from MySQL 8.0 to MySQL 8.4.
+
+**Impact:**
+
+* The use of non-unique or partial keys as foreign keys is now restricted by default
+* This may affect existing database structures that use non-standard foreign key configurations
+* Merchants must explicitly configure this setting to maintain compatibility
```

Comment:
```suggestion
* The use of non-unique or partial keys as foreign keys is now restricted by default.
* This can affect existing database structures that use non-standard foreign key configurations.
* Merchants must explicitly configure this setting to maintain compatibility.
```